### PR TITLE
Pattern prop for TextInput to give feedback about errors 

### PR DESCRIPTION
**Pull request by mrviniciux**
**Please check if the PR fulfills these requirements**

* [ x ] Tests for the changes have been added (for bug fixes / features)
* [ x ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* New prop named as "pattern" avaiable to handle with input expression errors and expressions which doesn't match the regex pattern;

* Supressed "console.error(err);" on idyll-document/src/utils/index.js as requested on issue (now we have a feature to give user feedback);

* Modified introduction.js on idyll-docs to give stylish to the new input with validation;

* Updated TextInput documentation.

**What is the current behavior?** (You can also link to an open issue here)
When the input was part of an evaluated expression and that expression was invalid an error occurs and also throwed on dev console leading to crash. See issue here: https://github.com/idyll-lang/idyll/issues/675

**What is the new behavior (if this is a feature change)?**
An error message is displayed within the input, so the user can have some feedback. Also the error on console is now suppressed.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

**Other information**:

Screenshot showing all the features commented above

### CODE REVIEW

**Comment by mathisonian** [comment]
This looks great @mrviniciux! Thanks for the PR and updating the doc strings as well. I've tested locally and it looks ready to merge, my only comment is that maybe the error message should be configurable? 

E.g. maybe we add something like:

```
[TextInput value:x pattern:... patternFailureMessage:"Input value doesn't match pattern." /]
```

So you can optionally customize that message. Open to suggestions for what that property should be called

**Review by mrviniciux — COMMENTED** [review]
> This looks great @mrviniciux! Thanks for the PR and updating the doc strings as well. I've tested locally and it looks ready to merge, my only comment is that maybe the error message should be configurable?
> 
> E.g. maybe we add something like:
> 
> ```
> [TextInput value:x pattern:... patternFailureMessage:"Input value doesn't match pattern." /]
> ```
> 
> So you can optionally customize that message. Open to suggestions for what that property should be called

Hey @mathisonian 

Thanks for the feedback. I've done the changes as requested and just changed the new prop to `patternMismatchMessage`, hope that works.

Can you review again?

File: packages/idyll-components/src/text-input.js [hunk 1]
```diff
@@ -1,25 +1,54 @@
...
+
+    this.props.updateProps({ value: value });
+
+    if (pattern) {
+      try {
+        if (!value.match(pattern)) {
+          throw new Error(patternMismatchMessage);
+        }
+
+        eval(value); //check if there's JS syntax errors
```
Comment (mathisonian) [inline packages/idyll-components/src/text-input.js:23]: Can we drop this `eval()`? I can imagine cases where a user could reasonably be expected to input something that isn't valid JS source code. 

e.g. something like

```
[var name:"username" value:"" /]

Enter your name: [TextInput value:username /]

Your name is [Display value:username /]
```

File: packages/idyll-docs/pages/docs/introduction.js [hunk 2]
```diff
@@ -223,6 +223,19 @@ const Introduction = ({ url }) => (
         max-width: 400px;
       }
 
+      .idyll-root input[type='text'].idyll-input-error {
+        border-color: red;
+      }
+
```
[the remark this answers is not in the document]
Comment (mathisonian) [reply packages/idyll-docs/pages/docs/introduction.js:229]: Let's add it to the built in Idyll themes as well. You could add the same CSS rule after related error styles [here](https://github.com/idyll-lang/idyll/blob/master/packages/idyll-themes/src/default/styles.js#L855-L858), [here](https://github.com/idyll-lang/idyll/blob/master/packages/idyll-themes/src/tufte/styles.js#L279-L282) and [here](https://github.com/idyll-lang/idyll/blob/master/packages/idyll-themes/src/github/styles.js#L737-L740)

**Comment by mathisonian** [comment]
Thanks for the updates @mrviniciux, things are looking good! I left a comment related to where to put styles, and added one question about the eval. I'm happy to merge once those are addressed.

**Comment by mrviniciux** [comment]
> Thanks for the updates @mrviniciux, things are looking good! I left a comment related to where to put styles, and added one question about the eval. I'm happy to merge once those are addressed.

Any time @mathisonian !

And I pushed a new commit with all requested alterations you mentioned. Hope that works

**Comment by mathisonian** [comment]
Looks great, thanks @mrviniciux! Merging this and will have it out in the next release on npm

**Comment by mrviniciux** [comment]
@mathisonian I think you closed the PR, is that right? And also thank you again for your fast feedback

**Comment by mathisonian** [comment]
D'oh, thanks for catching that 🙈. Actually merged now.

**Pull request merged**